diff --git a/Tools/CmdLine/IccSpecSepToTiff/iccSpecSepToTiff.cpp b/Tools/CmdLine/IccSpecSepToTiff/iccSpecSepToTiff.cpp
index 44ddd61..4747036 100644
--- a/Tools/CmdLine/IccSpecSepToTiff/iccSpecSepToTiff.cpp
+++ b/Tools/CmdLine/IccSpecSepToTiff/iccSpecSepToTiff.cpp
@@ -204,8 +204,8 @@ int main(int argc, char* argv[]) {
   long bytesPerSample = f->GetBitsPerSample()/8;
   
   // use unique_ptr to automatically free the buffers
-  std::unique_ptr<icUInt8Number> inbufffer( new icUInt8Number[ bytePerLine*nSamples ] );
-  std::unique_ptr<icUInt8Number> outbuffer( new icUInt8Number[ f->GetWidth() * bytesPerSample * nSamples ] );
+  std::unique_ptr<icUInt8Number[]> inbufffer( new icUInt8Number[ bytePerLine*nSamples ] );
+  std::unique_ptr<icUInt8Number[]> outbuffer( new icUInt8Number[ f->GetWidth() * bytesPerSample * nSamples ] );
   icUInt8Number *inbuf = inbufffer.get();
   icUInt8Number *outbuf = outbuffer.get();
 
@@ -225,7 +225,7 @@ int main(int argc, char* argv[]) {
   }
 
   // profile pointer lifetime needs to last until output file is written!
-  std::unique_ptr<unsigned char> destProfile;
+  std::unique_ptr<unsigned char[]> destProfile;
   if (argc>8) {
     CIccFileIO io;
     if (io.Open(argv[8], "rb")) {
