--- a/IccProfLib/IccPcc.cpp	2026-03-13 14:48:41.453026320 -0400
+++ b/IccProfLib/IccPcc.cpp	2026-03-13 14:48:57.491760594 -0400
@@ -293,6 +293,9 @@
   CIccMatrixMath *pAdjust=NULL, *pMtx;
   const CIccTagSpectralViewingConditions *pView = getPccViewingConditions();
 
+  if (!pView)
+    return NULL;
+
   icSpectralRange illumRange;
   const icFloatNumber *illum = pView->getIlluminant(illumRange);
 
@@ -320,7 +323,18 @@
 {
   if (pAppliedPCC) {
       const CIccTagSpectralViewingConditions *pView = pAppliedPCC->getPccViewingConditions();
-      if (bReflectance) {
+      if (!pView) {
+          m_pPCC = NULL;
+          m_pViewingConditions = NULL;
+          m_illuminantXYZ[0] = icD50XYZ[0];
+          m_illuminantXYZ[1] = icD50XYZ[1];
+          m_illuminantXYZ[2] = icD50XYZ[2];
+          m_illuminantXYZLum[0] = icD50XYZ[0];
+          m_illuminantXYZLum[1] = icD50XYZ[1];
+          m_illuminantXYZLum[2] = icD50XYZ[2];
+          m_bValidMediaXYZ = false;
+      }
+      else if (bReflectance) {
           m_pPCC = pAppliedPCC;
           m_pViewingConditions = NULL;
 
