diff --git a/IccProfLib/IccMatrixMath.cpp b/IccProfLib/IccMatrixMath.cpp
index 40a7218..e3a7f16 100644
--- a/IccProfLib/IccMatrixMath.cpp
+++ b/IccProfLib/IccMatrixMath.cpp
@@ -356,6 +356,9 @@ bool CIccMatrixMath::SetRange(const icSpectralRange &srcRange, const icSpectralR
   if (m_nRows != dstRange.steps || m_nCols != srcRange.steps)
     return false;
 
+  if (srcRange.steps < 2 || dstRange.steps < 2)
+    return false;
+
   icUInt16Number d;
   icFloatNumber srcStart = icF16toF(srcRange.start);
   icFloatNumber srcEnd = icF16toF(srcRange.end);
@@ -381,6 +384,8 @@ bool CIccMatrixMath::SetRange(const icSpectralRange &srcRange, const icSpectralR
     else {
       icUInt16Number p = (icUInt16Number)((w - srcStart) / srcScale);
       icFloatNumber p2 = (w - (srcStart + p * srcScale)) / srcScale;
+      if (p >= srcRange.steps - 1)
+        p = srcRange.steps - 2;
 
       if (p2<0.00001) {
         r[p] = 1.0f;
