diff --git a/IccProfLib/IccMpeCalc.cpp b/IccProfLib/IccMpeCalc.cpp
index b6818ba..18d20e0 100644
--- a/IccProfLib/IccMpeCalc.cpp
+++ b/IccProfLib/IccMpeCalc.cpp
@@ -950,7 +950,7 @@ public:
         || std::isnan(tempN) || std::isinf(tempN) || fabs(tempN) < epsilon)
         s[j] = 0.0;
       else
-        s[j] = temp - (icFloatNumber)((int)(temp / tempN))*tempN;
+        s[j] = (icFloatNumber)std::fmod((double)temp, (double)tempN);
     }
     OsShrinkArgs(n);
     return true;
