diff --git a/IccConnect/IccLibConnect/IccCmmConfig.cpp b/IccConnect/IccLibConnect/IccCmmConfig.cpp
--- a/IccConnect/IccLibConnect/IccCmmConfig.cpp
+++ b/IccConnect/IccLibConnect/IccCmmConfig.cpp
@@ -1497,6 +1497,7 @@ bool CIccCfgColorData::fromIt8(const char* filename, bool bReset)
         if (space != lastSpace) {
           lastSpace = space;
           icValueVector val(4);
           val[0].space = icSigCmykData;
+          samples.push_back(val);
           spaces.push_back(space);
         }
