diff --git a/IccProfLib/IccMpeCalc.cpp b/IccProfLib/IccMpeCalc.cpp
index 289e401..e893cfe 100644
--- a/IccProfLib/IccMpeCalc.cpp
+++ b/IccProfLib/IccMpeCalc.cpp
@@ -1265,13 +1265,9 @@ public:
     for (j=0; j<n; j++) {
       icFloatNumber temp = s[j];
       if (std::isnan(temp))
-        temp = 0.0;
-      else if (std::isinf(temp))
-        temp = 10000.0;          // value chosen arbitrarily to not overflow int
-      if (temp < 0.0)
-        s[j] = icFloatNumber((int)(temp-0.5));
+        s[j] = 0.0;
       else
-        s[j] = icFloatNumber((int)(temp+0.5));
+        s[j] = (icFloatNumber)std::round((double)temp);
     }
     return true;
   }
