--- a/IccConnect/IccLibConnect/IccCmmConfig.cpp
+++ b/IccConnect/IccLibConnect/IccCmmConfig.cpp
@@ -838,7 +838,7 @@
 
 CIccCfgPccWeight::CIccCfgPccWeight()
 {
-
+  m_dWeight = 0.0f;
 }
 
 CIccCfgPccWeight::~CIccCfgPccWeight() {
@@ -884,10 +884,22 @@
 
 CIccCfgSearchApply::CIccCfgSearchApply()
 {
+  m_intentInitial = icUnknownIntent;
+  m_transformInitial = icXformLutColor;
+  m_useD2BxB2DxInitial = false;
+  m_adjustPcsLuminanceInitial = false;
+  m_useV5SubProfileInitial = false;
+  m_interpolationInitial = icInterpTetrahedral;
 }
 
 void CIccCfgSearchApply::reset()
 {
+  m_intentInitial = icUnknownIntent;
+  m_transformInitial = icXformLutColor;
+  m_useD2BxB2DxInitial = false;
+  m_adjustPcsLuminanceInitial = false;
+  m_useV5SubProfileInitial = false;
+  m_interpolationInitial = icInterpTetrahedral;
   m_pccWeights.clear();
   m_profiles.clear();
 }
