diff --git a/IccProfLib/IccUtil.cpp b/IccProfLib/IccUtil.cpp
index 4d2c0b0..43d6d67 100644
--- a/IccProfLib/IccUtil.cpp
+++ b/IccProfLib/IccUtil.cpp
@@ -1085,7 +1085,7 @@ const icChar *icGetSig(icChar *pBuf, size_t bufSize, icUInt32Number nSig, bool b
       bGetHexVal = true;
     }
     pBuf[i]=c;
-    sig <<=8;
+    if (i < 4) sig <<=8;
   }
 
   if (bGetHexVal)
@@ -1127,7 +1127,7 @@ const icChar *icGetSigStr(icChar *pBuf, size_t bufSize, icUInt32Number nSig)
       bGetHexVal = true;
     }
     pBuf[i]=c;
-    sig <<=8;
+    if (i < 3) sig <<=8;
   }
 
   if (bGetHexVal)
@@ -1184,7 +1184,7 @@ const icChar *icGetColorSig(icChar *pBuf, size_t bufSize, icUInt32Number nSig, b
           bNeedHexVal = true;
         }
         pBuf[i]=c;
-        sig <<=8;
+        if (i < 4) sig <<=8;
       }
 
       if (bGetHexVal)
@@ -1250,7 +1250,7 @@ const icChar *icGetColorSigStr(icChar *pBuf, size_t bufSize, icUInt32Number nSig
             bGetHexVal = true;
           }
           pBuf[i]=c;
-          sig <<=8;
+          if (i < 3) sig <<=8;
         }
 
         if (bGetHexVal)
