diff --git a/IccProfLib/IccUtil.cpp b/IccProfLib/IccUtil.cpp
index b0e7513..d627a58 100644
--- a/IccProfLib/IccUtil.cpp
+++ b/IccProfLib/IccUtil.cpp
@@ -1084,7 +1084,7 @@ const icChar *icGetSig(icChar *pBuf, size_t bufSize, icUInt32Number nSig, bool b
       c='?';
       bGetHexVal = true;
     }
-    pBuf[i]=c;
+    pBuf[i]=(icChar)c;
   }
 
   if (bGetHexVal)
@@ -1124,7 +1124,7 @@ const icChar *icGetSigStr(icChar *pBuf, size_t bufSize, icUInt32Number nSig)
       c='?';
       bGetHexVal = true;
     }
-    pBuf[i]=c;
+    pBuf[i]=(icChar)c;
   }
 
   if (bGetHexVal)
@@ -1179,7 +1179,7 @@ const icChar *icGetColorSig(icChar *pBuf, size_t bufSize, icUInt32Number nSig, b
           c = '?';
           bNeedHexVal = true;
         }
-        pBuf[i]=c;
+        pBuf[i]=(icChar)c;
       }
 
       if (bGetHexVal)
@@ -1243,7 +1243,7 @@ const icChar *icGetColorSigStr(icChar *pBuf, size_t bufSize, icUInt32Number nSig
             c='?';
             bGetHexVal = true;
           }
-          pBuf[i]=c;
+          pBuf[i]=(icChar)c;
         }
 
         if (bGetHexVal)
