diff --git a/IccProfLib/IccIO.cpp b/IccProfLib/IccIO.cpp
index 0dbbcbd..0d60ea8 100644
--- a/IccProfLib/IccIO.cpp
+++ b/IccProfLib/IccIO.cpp
@@ -612,6 +612,9 @@ size_t CIccEmbedIO::Read8(void *pBuf, size_t nNum)
     size_t nPos = m_pIO->Tell();
     size_t nOffset = nPos - m_nStartPos;
 
+    if (nOffset >= m_nSize)
+      return 0;
+
     if (nOffset + nNum > m_nSize)
       nNum = m_nSize - nOffset;
   }
@@ -782,6 +785,8 @@ size_t CIccMemIO::Read8(void *pBuf, size_t nNum)
 {
   if (!m_pData || !pBuf)
     return 0;
+  if (m_nPos > m_nSize)
+    return 0;
   if (nNum > 0) {
       nNum = __min((m_nSize - m_nPos), nNum);
       memcpy(pBuf, m_pData + m_nPos, nNum);
