--- a/IccProfLib/IccMpeBasic.cpp
+++ b/IccProfLib/IccMpeBasic.cpp
@@ -1940,7 +1940,9 @@
   m_nCount = 0;
   m_pSamples = 0;
 
+  m_storageType = icValueTypeFloat32;
   m_extensionType = icClipSingleSampledCurve;
+  m_nDesiredSize = 0;
 
   if (first < last) {
     m_firstEntry = first;
@@ -1951,6 +1953,9 @@
     m_lastEntry = first;
   }
 
+  m_range = 0;
+  m_last = 0;
+
   m_loIntercept = 0;
   m_loSlope = 0;
   m_hiIntercept = 1.0;
@@ -4923,6 +4928,8 @@
   m_size = 0;
   m_pMatrix = NULL;
   m_pConstants = NULL;
+  m_type = icOtherMatrix;
+  m_bApplyConstants = false;
 }
 
 
