diff --git a/IccConnect/IccLibConnect/IccCmmConfig.cpp b/IccConnect/IccLibConnect/IccCmmConfig.cpp
index 3bc7292..87647d9 100644
--- a/IccConnect/IccLibConnect/IccCmmConfig.cpp
+++ b/IccConnect/IccLibConnect/IccCmmConfig.cpp
@@ -561,6 +561,7 @@ void CIccCfgProfile::reset()
   m_useBPC = false;
   m_useHToS = false;
   m_useV5SubProfile = false;
+  m_useEmbedded = false;
   m_interpolation = icInterpTetrahedral;
 }
 
@@ -1374,7 +1375,7 @@ bool CIccCfgColorData::fromLegacy(const char* filename, bool bReset)
   }
 
   InputData.getline(tempBuf, tempBufSize);
-  sscanf(tempBuf, "%s", tempBuf);
+  sscanf(tempBuf, "%19999s", tempBuf);
 
   //Setup source encoding
   m_encoding = CIccCmm::GetFloatColorEncoding(tempBuf);
@@ -2046,7 +2047,7 @@ bool CIccCfgColorData::toIt8(const char* filename, icUInt8Number nDigits, icUInt
   fprintf(f, "BEGIN_DATA_FORMAT\n");
   fprintf(f, "%s\n", dataFormat.c_str());
   fprintf(f, "END_DATA_FORMAT\n");
-  fprintf(f, "NUMBER_OF_SETS\t%u\n", (int)m_data.size());
+  fprintf(f, "NUMBER_OF_SETS\t%d\n", (int)m_data.size());
 
   CIccCfgDataEntry blank;
   const size_t bufSize = 256;
