diff --git a/IccProfLib/IccTagLut.h b/IccProfLib/IccTagLut.h
index 0e2b5f6..e9c5f0e 100644
--- a/IccProfLib/IccTagLut.h
+++ b/IccProfLib/IccTagLut.h
@@ -640,12 +640,12 @@ public:
     return m_NumberOfTriangles;
   }	
 
-  icInt16Number getNumPCSChannels() 
+  icUInt16Number getNumPCSChannels() 
   {
     return m_nPCSChannels;
   }
 
-  icInt16Number getNumDeviceChannels() 
+  icUInt16Number getNumDeviceChannels() 
   {
     return m_nDeviceChannels;
   }		
@@ -653,8 +653,8 @@ public:
 protected:
   icInt32Number m_NumberOfVertices;
   icInt32Number m_NumberOfTriangles;
-  icInt16Number m_nPCSChannels;
-  icInt16Number m_nDeviceChannels;
+  icUInt16Number m_nPCSChannels;
+  icUInt16Number m_nDeviceChannels;
   icFloatNumber* m_PCSValues;
   icFloatNumber* m_DeviceValues;
   icGamutBoundaryTriangle* m_Triangles;
