diff --git a/IccProfLib/IccCmm.cpp b/IccProfLib/IccCmm.cpp
--- a/IccProfLib/IccCmm.cpp
+++ b/IccProfLib/IccCmm.cpp
@@ -8656,7 +8656,7 @@ icStatusCMM CIccCmm::AddXform(CIccProfile &Profile,
     return icCmmStatAllocErr;
 
   //borrow the caller's AttachIO to perform the AddXform
-  pProfile->CopyAttach(&Profile);
+  pProfile->CopyAttach(&Profile, true);
 
   // CFL-078: Save deviceClass before AddXform - cenc ownership transfer
   icProfileClassSignature savedClass = pProfile->m_Header.deviceClass;
