--- a/IccProfLib/IccTagComposite.cpp	2026-02-28 21:14:23.135086464 -0500
+++ IccProfLib/IccTagComposite.cpp	2026-02-28 21:14:23.135086464 -0500
@@ -1037,7 +1037,7 @@
   m_nSize = 0;
 
   if (tagAry.m_nSize) {
-    m_TagVals = new IccTagPtr[tagAry.m_nSize];
+    m_TagVals = (IccTagPtr*)calloc(tagAry.m_nSize, sizeof(IccTagPtr));
 
     icUInt32Number i;
     for (i=0; i<tagAry.m_nSize; i++) {
@@ -1074,7 +1074,7 @@
   Cleanup();
 
   if (tagAry.m_nSize) {
-    m_TagVals = new IccTagPtr[tagAry.m_nSize];
+    m_TagVals = (IccTagPtr*)calloc(tagAry.m_nSize, sizeof(IccTagPtr));
 
     icUInt32Number i;
     for (i=0; i<tagAry.m_nSize; i++) {
