diff --git a/IccProfLib/IccMpeCalc.cpp b/IccProfLib/IccMpeCalc.cpp
index f49da45..f2b1e90 100644
--- a/IccProfLib/IccMpeCalc.cpp
+++ b/IccProfLib/IccMpeCalc.cpp
@@ -3474,11 +3474,13 @@ bool CIccCalculatorFunc::Read(icUInt32Number size, CIccIO *pIO)
     return false;
   }
 
-  icChannelFuncSignature sig;
+  icUInt32Number rawChSig;
 
-  if (!pIO->Read32(&sig))
+  if (!pIO->Read32(&rawChSig))
     return false;
 
+  icChannelFuncSignature sig = static_cast<icChannelFuncSignature>(rawChSig);
+
   if (sig!= GetType())
     return false;
 
