--- a/IccProfLib/IccMpeCalc.cpp	2026-02-24 16:24:04.829464355 -0500
+++ IccProfLib/IccMpeCalc.cpp	2026-02-24 16:24:04.833450370 -0500
@@ -3660,9 +3660,9 @@
 
 
   icUInt32Number i, n, pos;
-  for (n=0; n<nOps && ops[n+1].sig==icSigCaseOp; n++);
+  for (n=0; n+1<nOps && ops[n+1].sig==icSigCaseOp; n++);
   ops->extra=n;
-  if (ops[n+1].sig==icSigDefaultOp) {
+  if (n+1<nOps && ops[n+1].sig==icSigDefaultOp) {
     n++;
   }
   pos = n;
