--- a/IccProfLib/IccMpeCalc.cpp	2026-02-24 16:47:35.394174624 -0500
+++ IccProfLib/IccMpeCalc.cpp	2026-02-24 16:47:44.901227925 -0500
@@ -329,18 +329,18 @@
 public:
   virtual bool Exec(SIccCalcOp *op, SIccOpState &os)
   {
-    icSigCmmEnvVar sig = (icSigCmmEnvVar) op->data.size;
+    icUInt32Number rawSig = op->data.size;
     icFloatNumber val=0.0;
 
-    if (sig==icSigTrueVar) {
+    if (rawSig==static_cast<icUInt32Number>(icSigTrueVar)) {
       OsPushArg((icFloatNumber)1.0);
       OsPushArg((icFloatNumber)1.0);
     }
-    else if (sig==icSigNotDefVar) {
+    else if (rawSig==static_cast<icUInt32Number>(icSigNotDefVar)) {
       OsPushArg((icFloatNumber)0.0);
       OsPushArg((icFloatNumber)0.0);
     }
-    else if (os.pApply->GetEnvVar(sig, val)) {
+    else if (os.pApply->GetEnvVar(static_cast<icSigCmmEnvVar>(rawSig), val)) {
       OsPushArg((icFloatNumber)val);
       OsPushArg((icFloatNumber)1.0);
     }
