diff --git a/IccProfLib/IccTagLut.cpp b/IccProfLib/IccTagLut.cpp
index 6328394..b60b7a4 100644
--- a/IccProfLib/IccTagLut.cpp
+++ b/IccProfLib/IccTagLut.cpp
@@ -5660,7 +5660,7 @@ bool CIccTagGamutBoundaryDesc::Read(icUInt32Number size, CIccIO *pIO)
 	}
 	m_Triangles = new icGamutBoundaryTriangle[m_NumberOfTriangles];
 	
-	icUInt32Number nNum32 = m_NumberOfTriangles*3;
+	icUInt32Number nNum32 = (icUInt32Number)m_NumberOfTriangles*3;
 	
 	if (pIO->Read32(m_Triangles, nNum32)!=nNum32)
 		return false;		
@@ -5717,7 +5717,7 @@ bool CIccTagGamutBoundaryDesc::Write(CIccIO *pIO)
 		!pIO->Write32(&m_NumberOfTriangles))
 		return false;	
 	
-	icUInt32Number nNum32 = m_NumberOfTriangles*3;
+	icUInt32Number nNum32 = (icUInt32Number)m_NumberOfTriangles*3;
 	
 	if (pIO->Write32(m_Triangles, nNum32)!=nNum32)
 		return false;	
