diff --git a/IccProfLib/IccTagLut.h b/IccProfLib/IccTagLut.h
index 8dac8f4..fd1f8cb 100644
--- a/IccProfLib/IccTagLut.h
+++ b/IccProfLib/IccTagLut.h
@@ -139,8 +139,15 @@ public:
   virtual bool Read(icUInt32Number size, CIccIO *pIO);
   virtual bool Write(CIccIO *pIO);
 
-  icFloatNumber &operator[](icUInt32Number index) {return m_Curve[index];}
-  icFloatNumber *GetData(icUInt32Number index) {return &m_Curve[index];}
+  icFloatNumber &operator[](icUInt32Number index) {
+    static icFloatNumber dummy = 0;
+    if (!m_Curve) return dummy;
+    return m_Curve[index];
+  }
+  icFloatNumber *GetData(icUInt32Number index) {
+    if (!m_Curve) return NULL;
+    return &m_Curve[index];
+  }
   icUInt32Number GetSize() const { return m_nSize; }
   bool SetSize(icUInt32Number nSize, icTagCurveSizeInit nSizeOpt=icInitZero);
   bool SetGamma(icFloatNumber gamma);
