diff --git a/IccProfLib/IccMpeCalc.cpp b/IccProfLib/IccMpeCalc.cpp
index 1b6f1be..b6818ba 100644
--- a/IccProfLib/IccMpeCalc.cpp
+++ b/IccProfLib/IccMpeCalc.cpp
@@ -3774,6 +3774,9 @@ bool CIccCalculatorFunc::ApplySequence(CIccApplyMpeCalculator *pApply, icUInt32N
           if (os.idx+1 + ops[nDefOff].extra >= nOps)
             return false;
 
+          if (os.idx+1 + ops[nDefOff].extra + ops[nDefOff].data.size > nOps)
+            return false;
+
           if (!ApplySequence(pApply, ops[nDefOff].data.size, &ops[os.idx+1 + ops[nDefOff].extra]))
             break;
         }
@@ -3784,6 +3787,9 @@ bool CIccCalculatorFunc::ApplySequence(CIccApplyMpeCalculator *pApply, icUInt32N
         if (nOff >= nOps) 
           return false;
 
+        if (os.idx+1 + ops[nOff].extra + ops[nOff].data.size > nOps)
+          return false;
+
         if (!ApplySequence(pApply, ops[nOff].data.size, &ops[os.idx+1 + ops[nOff].extra]))
           break;
       }
@@ -3797,6 +3803,9 @@ bool CIccCalculatorFunc::ApplySequence(CIccApplyMpeCalculator *pApply, icUInt32N
       else if (op->extra) {
         unsigned long nOff = os.idx + op->extra;
 
+        if (nOff >= nOps)
+          return false;
+
         if (os.idx+1 + ops[nOff].extra + ops[nOff].data.size > nOps)
           return false;
 
