diff --git a/Tools/CmdLine/IccCommon/IccCmmConfig.cpp b/Tools/CmdLine/IccCommon/IccCmmConfig.cpp
index c944c5f..876967f 100644
--- a/Tools/CmdLine/IccCommon/IccCmmConfig.cpp
+++ b/Tools/CmdLine/IccCommon/IccCmmConfig.cpp
@@ -698,10 +698,15 @@ void CIccCfgProfile::toJson(json& j) const
   if (m_useV5SubProfile)
     j["useV5SubProfile"] = m_useV5SubProfile;
   int i;
+  for (i = 0; icTranNames[i]; i++)
+    if (icTranValues[i] == m_transform)
+      break;
+  if (icTranNames[i])
+    j["transform"] = icTranNames[i];
   for (i = 0; icInterpNames[i]; i++)
     if (icInterpValues[i] == m_interpolation)
       break;
-  if (icInterpValues[i] == icInterpLinear)
+  if (icInterpNames[i])
     j["interpolation"] = icInterpNames[i];
 }
 
