diff --git a/IccProfLib/IccCmm.cpp b/IccProfLib/IccCmm.cpp
index bf22ca3..3cb53b3 100644
--- a/IccProfLib/IccCmm.cpp
+++ b/IccProfLib/IccCmm.cpp
@@ -3053,6 +3053,8 @@ icStatusCMM CIccPcsXform::pushXYZConvert(CIccXform *pFromXform, CIccXform *pToXf
 void CIccPcsXform::pushXYZNormalize(IIccProfileConnectionConditions *pPcc, const icSpectralRange &srcRange, const icSpectralRange &dstRange)
 {
   const CIccTagSpectralViewingConditions *pView = pPcc->getPccViewingConditions();
+  if (!pView)
+    return;
   CIccPcsXform tmp;
 
   icSpectralRange illuminantRange;
