diff --git a/IccProfLib/IccMpeCalc.cpp b/IccProfLib/IccMpeCalc.cpp
index 6d868ca..9b49de4 100644
--- a/IccProfLib/IccMpeCalc.cpp
+++ b/IccProfLib/IccMpeCalc.cpp
@@ -4082,8 +4082,13 @@ bool CIccCalculatorFunc::SequenceNeedTempReset(SIccCalcOp *op, icUInt32Number nO
 * 
 * Return: 
 ******************************************************************************/
-int CIccCalculatorFunc::CheckUnderflowOverflow(SIccCalcOp *op, icUInt32Number nOps, int nArgs, bool bCheckUnderflow, std::string &sReport) const
+int CIccCalculatorFunc::CheckUnderflowOverflow(SIccCalcOp *op, icUInt32Number nOps, int nArgs, bool bCheckUnderflow, std::string &sReport, icUInt32Number nRecurseDepth) const
 {
+  static const icUInt32Number kMaxRecurseDepth = 100;
+
+  if (nRecurseDepth > kMaxRecurseDepth)
+    return -1;
+
   icUInt32Number i, p;
   int nIfArgs, nElseArgs, nSelArgs, nCaseArgs;
 
@@ -4131,7 +4136,7 @@ int CIccCalculatorFunc::CheckUnderflowOverflow(SIccCalcOp *op, icUInt32Number nO
         p = i+2; 
         if (p > nOps)
           return -1;
-        nIfArgs = CheckUnderflowOverflow(&op[p], icIntMin(nOps-p, op[i].data.size), nArgs, bCheckUnderflow, sReport);
+        nIfArgs = CheckUnderflowOverflow(&op[p], icIntMin(nOps-p, op[i].data.size), nArgs, bCheckUnderflow, sReport, nRecurseDepth + 1);
         if (nIfArgs<0)
           return -1;
         incI =op[i].data.size;
@@ -4139,7 +4144,7 @@ int CIccCalculatorFunc::CheckUnderflowOverflow(SIccCalcOp *op, icUInt32Number nO
         p = i+2+op[i].data.size;
         if (p > nOps)
           return -1;
-        nElseArgs = CheckUnderflowOverflow(&op[p], icIntMin(nOps-p, op[i+1].data.size), nArgs, bCheckUnderflow, sReport);
+        nElseArgs = CheckUnderflowOverflow(&op[p], icIntMin(nOps-p, op[i+1].data.size), nArgs, bCheckUnderflow, sReport, nRecurseDepth + 1);
         if (nElseArgs<0)
           return -1;
         incI += op[i+1].data.size;
@@ -4152,7 +4157,7 @@ int CIccCalculatorFunc::CheckUnderflowOverflow(SIccCalcOp *op, icUInt32Number nO
         p = i+1; 
         if (p > nOps)
           return -1;
-        nIfArgs = CheckUnderflowOverflow(&op[p], icIntMin(nOps-p, op[i].data.size), nArgs, bCheckUnderflow, sReport);
+        nIfArgs = CheckUnderflowOverflow(&op[p], icIntMin(nOps-p, op[i].data.size), nArgs, bCheckUnderflow, sReport, nRecurseDepth + 1);
         if (nIfArgs<0)
           return -1;
         nArgs = bCheckUnderflow ? icIntMin(nArgs, nIfArgs) : icIntMax(nArgs, nIfArgs);
@@ -4188,7 +4193,7 @@ int CIccCalculatorFunc::CheckUnderflowOverflow(SIccCalcOp *op, icUInt32Number nO
 
         if (pos > nOps)
           return -1;
-        nCaseArgs = CheckUnderflowOverflow(&op[pos], icIntMin(nOps-pos, len), nArgs, bCheckUnderflow, sReport);
+        nCaseArgs = CheckUnderflowOverflow(&op[pos], icIntMin(nOps-pos, len), nArgs, bCheckUnderflow, sReport, nRecurseDepth + 1);
         if (nCaseArgs<0)
           return -1;
 
diff --git a/IccProfLib/IccMpeCalc.h b/IccProfLib/IccMpeCalc.h
index ad1e40c..3e123ed 100644
--- a/IccProfLib/IccMpeCalc.h
+++ b/IccProfLib/IccMpeCalc.h
@@ -388,7 +388,7 @@ public:
   icFuncParseStatus SetFunction(CIccCalcOpList &opList, std::string &sReport);
 
   icUInt32Number GetMaxTemp() const;
-  int CheckUnderflowOverflow(SIccCalcOp *op, icUInt32Number nOps, int nArgs, bool bCheckUnderflow, std::string &sReport) const;
+  int CheckUnderflowOverflow(SIccCalcOp *op, icUInt32Number nOps, int nArgs, bool bCheckUnderflow, std::string &sReport, icUInt32Number nRecurseDepth = 0) const;
   icFuncParseStatus DoesStackUnderflowOverflow(std::string &sReport) const;
   bool HasValidOperations(std::string &sReport) const;
   bool HasUnsupportedOperations(std::string &sReport, const CIccProfile *pProfile) const;
