diff --git a/IccProfLib/IccCmm.cpp b/IccProfLib/IccCmm.cpp
index 07a5f24..90bc619 100644
--- a/IccProfLib/IccCmm.cpp
+++ b/IccProfLib/IccCmm.cpp
@@ -2512,12 +2512,18 @@ icStatusCMM CIccPcsXform::Optimize()
         if (!ptr.ptr->isIdentity()) {
           newSteps.push_back(ptr);
         }
+        else {
+          delete ptr.ptr;
+        }
         ptr.ptr = next->ptr;
       }
     }
     if (!ptr.ptr->isIdentity()) {
       newSteps.push_back(ptr);
     }
+    else {
+      delete ptr.ptr;
+    }
 
     steps = newSteps;
 
@@ -6539,6 +6545,18 @@ void CIccXformNDLut::Apply(CIccApplyXform* pApply, icFloatNumber *DstPixel, cons
 
     if (m_pTag->m_CLUT) {
       switch(nInput) {
+      case 1:
+        m_pTag->m_CLUT->Interp1d(Pixel, Pixel);
+        break;
+      case 2:
+        m_pTag->m_CLUT->Interp2d(Pixel, Pixel);
+        break;
+      case 3:
+        m_pTag->m_CLUT->Interp3d(Pixel, Pixel);
+        break;
+      case 4:
+        m_pTag->m_CLUT->Interp4d(Pixel, Pixel);
+        break;
       case 5:
         m_pTag->m_CLUT->Interp5d(Pixel, Pixel);
         break;
@@ -6548,7 +6566,8 @@ void CIccXformNDLut::Apply(CIccApplyXform* pApply, icFloatNumber *DstPixel, cons
       default:
         {
           CIccApplyNDLutXform* pNDApply = (CIccApplyNDLutXform*)pApply;
-          m_pTag->m_CLUT->InterpND(Pixel, Pixel, pNDApply->m_pApply);
+          if (pNDApply && pNDApply->m_pApply)
+            m_pTag->m_CLUT->InterpND(Pixel, Pixel, pNDApply->m_pApply);
           break;
         }
       }
@@ -6569,6 +6588,18 @@ void CIccXformNDLut::Apply(CIccApplyXform* pApply, icFloatNumber *DstPixel, cons
 
     if (m_pTag->m_CLUT) {
       switch(m_nNumInput) {
+      case 1:
+        m_pTag->m_CLUT->Interp1d(Pixel, Pixel);
+        break;
+      case 2:
+        m_pTag->m_CLUT->Interp2d(Pixel, Pixel);
+        break;
+      case 3:
+        m_pTag->m_CLUT->Interp3d(Pixel, Pixel);
+        break;
+      case 4:
+        m_pTag->m_CLUT->Interp4d(Pixel, Pixel);
+        break;
       case 5:
         m_pTag->m_CLUT->Interp5d(Pixel, Pixel);
         break;
@@ -6578,10 +6609,10 @@ void CIccXformNDLut::Apply(CIccApplyXform* pApply, icFloatNumber *DstPixel, cons
       default:
       {
         CIccApplyNDLutXform* pNDApply = (CIccApplyNDLutXform*)pApply;
-        m_pTag->m_CLUT->InterpND(Pixel, Pixel, pNDApply->m_pApply);
+        if (pNDApply && pNDApply->m_pApply)
+          m_pTag->m_CLUT->InterpND(Pixel, Pixel, pNDApply->m_pApply);
         break;
       }
-      break;
       }
     }
 
@@ -8579,8 +8610,10 @@ icStatusCMM CIccCmm::CheckPCSConnections(bool bUsePCSConversions/*=false*/)
 
         rv = pPcs->Connect(last->ptr, next->ptr);
 
-        if (rv!=icCmmStatOk && rv!=icCmmStatIdentityXform)
+        if (rv!=icCmmStatOk && rv!=icCmmStatIdentityXform) {
+          delete pPcs;
           return rv;
+        }
 
         if (rv!=icCmmStatIdentityXform) {
           ptr.ptr = pPcs;
