--- a/IccProfLib/IccTagLut.cpp
+++ b/IccProfLib/IccTagLut.cpp
@@ -891,7 +891,7 @@
 
 case 0x0003:
   if (m_nNumParam < 5) { sDescription += "  Insufficient parameters\n"; return; }
-  snprintf(buf, bufSize, "Y = %lf * X   when (X < %.4lf)\n",
+  snprintf(buf, bufSize, "Y = %.4lf * X   when (X < %.4lf)\n",
     m_dParam[3], m_dParam[4]);
   sDescription += buf;
 
@@ -903,7 +903,7 @@
 
 case 0x0004:
   if (m_nNumParam < 7) { sDescription += "  Insufficient parameters\n"; return; }
-  snprintf(buf, bufSize, "Y = %lf * X + %.4lf  when (X < %.4lf)\n",
+  snprintf(buf, bufSize, "Y = %.4lf * X + %.4lf  when (X < %.4lf)\n",
     m_dParam[3], m_dParam[6], m_dParam[4]);
   sDescription += buf;
 
