--- a/Tools/CmdLine/IccCommon/IccCmmConfig.cpp
+++ b/Tools/CmdLine/IccCommon/IccCmmConfig.cpp
@@ -1550,10 +1550,10 @@
       }
       else {
       int nColor = -1;
-      if ( (sscanf(szFmt, "%uCOLOR_", &nColor) == 1) && nColor >= 1) {  // sscanf can also return EOF(-1)
+      if ( (sscanf(szFmt, "%dCOLOR_", &nColor) == 1) && nColor >= 1) {  // sscanf can also return EOF(-1)
         const size_t bufSize = 30;
         char buf[bufSize];
-        snprintf(buf, bufSize, "%uCOLOR", nColor);
+        snprintf(buf, bufSize, "%dCOLOR", nColor);
         space += buf;
 
         if (space != lastSpace) {
