diff --git a/IccProfLib/IccCmm.cpp b/IccProfLib/IccCmm.cpp
index 7be4a0a..07a5f24 100644
--- a/IccProfLib/IccCmm.cpp
+++ b/IccProfLib/IccCmm.cpp
@@ -2946,16 +2946,18 @@ icStatusCMM CIccPcsXform::pushXYZConvert(CIccXform *pFromXform, CIccXform *pToXf
         if (pElem->GetType()==icSigMatrixElemType) {
           CIccMpeMatrix *pMatElem = (CIccMpeMatrix*)pElem;
 
-          icFloatNumber *pMat = pMatElem->GetMatrix();
-          icFloatNumber *pOffset = pMatElem->GetConstants();
+          if (pMatElem->NumInputChannels()==3 && pMatElem->NumOutputChannels()==3) {
+            icFloatNumber *pMat = pMatElem->GetMatrix();
+            icFloatNumber *pOffset = pMatElem->GetConstants();
 
-          if (pMat && (!pOffset || (pOffset[0]==0.0 && pOffset[1]==0.0 && pOffset[2]==0.0))) {
-            CIccPcsStepMatrix *pStepMtx = new CIccPcsStepMatrix(3, 3);
+            if (pMat && (!pOffset || (pOffset[0]==0.0 && pOffset[1]==0.0 && pOffset[2]==0.0))) {
+              CIccPcsStepMatrix *pStepMtx = new CIccPcsStepMatrix(3, 3);
 
-            if (pStepMtx ) {
-              memcpy(pStepMtx->entry(0,0), pMat, 9*sizeof(icFloatNumber));
+              if (pStepMtx ) {
+                memcpy(pStepMtx->entry(0,0), pMat, 9*sizeof(icFloatNumber));
+              }
+              ptr.ptr = pStepMtx;
             }
-            ptr.ptr = pStepMtx;
           }
         }
       }
@@ -2994,16 +2996,18 @@ icStatusCMM CIccPcsXform::pushXYZConvert(CIccXform *pFromXform, CIccXform *pToXf
         if (pElem->GetType()==icSigMatrixElemType) {
           CIccMpeMatrix *pMatElem = (CIccMpeMatrix*)pElem;
 
-          icFloatNumber *pMat = pMatElem->GetMatrix();
-          icFloatNumber *pOffset = pMatElem->GetConstants();
+          if (pMatElem->NumInputChannels()==3 && pMatElem->NumOutputChannels()==3) {
+            icFloatNumber *pMat = pMatElem->GetMatrix();
+            icFloatNumber *pOffset = pMatElem->GetConstants();
 
-          if (pMat && (!pOffset || (pOffset[0]==0.0 && pOffset[1]==0.0 && pOffset[2]==0.0))) {
-            CIccPcsStepMatrix *pStepMtx = new CIccPcsStepMatrix(3, 3);
+            if (pMat && (!pOffset || (pOffset[0]==0.0 && pOffset[1]==0.0 && pOffset[2]==0.0))) {
+              CIccPcsStepMatrix *pStepMtx = new CIccPcsStepMatrix(3, 3);
 
-            if (pStepMtx ) {
-              memcpy(pStepMtx->entry(0,0), pMat, 9*sizeof(icFloatNumber));
+              if (pStepMtx ) {
+                memcpy(pStepMtx->entry(0,0), pMat, 9*sizeof(icFloatNumber));
+              }
+              ptr.ptr = pStepMtx;
             }
-            ptr.ptr = pStepMtx;
           }
         }
       }
