diff --git a/IccProfLib/IccMpeCalc.cpp b/IccProfLib/IccMpeCalc.cpp
index d3b44f0..53084a8 100644
--- a/IccProfLib/IccMpeCalc.cpp
+++ b/IccProfLib/IccMpeCalc.cpp
@@ -3251,7 +3251,7 @@ const char *CIccCalculatorFunc::ParseFuncDef(const char *szFuncDef, CIccCalcOpLi
           icSigCmmEnvVar envSig;
           if (!scan.GetEnvSig(envSig))
             return NULL;
-          op.data.size = (icUInt32Number)envSig;
+          memcpy(&op.data.size, &envSig, sizeof(op.data.size));
           scanList.push_back(op);
         }
         break;
