diff --git a/IccProfLib/IccMpeCalc.cpp b/IccProfLib/IccMpeCalc.cpp
index ee49926..f6de54d 100644
--- a/IccProfLib/IccMpeCalc.cpp
+++ b/IccProfLib/IccMpeCalc.cpp
@@ -4742,7 +4742,7 @@ bool CIccMpeCalculator::Read(icUInt32Number size, CIccIO *pIO)
       // No, you may not make circular references in the tag,
       // reference back into the tag header,
       // or try to load more data than we have available.
-      if ( (pos->offset < headerSize) || ((pos->offset + pos->size) > size) ) {
+      if ( (pos->offset < headerSize) || ( pos->size > size || pos->offset > size - pos->size ) ) {
         free(posvals);
         return false;
       }
@@ -4774,7 +4774,7 @@ bool CIccMpeCalculator::Read(icUInt32Number size, CIccIO *pIO)
   pos = posvals;
   
   // overreading, references into the header, or not having a calc func would be bad
-  if ( !m_calcFunc || (pos->offset < headerSize) || ((pos->offset + pos->size) > size) ) {
+  if ( !m_calcFunc || (pos->offset < headerSize) || ( pos->size > size || pos->offset > size - pos->size ) ) {
     free(posvals);
     return false;
   }
diff --git a/IccProfLib/IccProfile.cpp b/IccProfLib/IccProfile.cpp
index 4362b23..f7e68d1 100644
--- a/IccProfLib/IccProfile.cpp
+++ b/IccProfLib/IccProfile.cpp
@@ -1308,7 +1308,7 @@ bool CIccProfile::LoadTag(IccTagEntry *pTagEntry, CIccIO *pIO, bool bReadAll/*=f
   
   // if the tag claims to be longer than the actual file, return an error
   // NOTE - ccox - it would be nice to cache the file length instead of calculating it per tag
-  if ( (pTagEntry->TagInfo.offset + pTagEntry->TagInfo.size) > pIO->GetLength())
+  if ( pTagEntry->TagInfo.size > pIO->GetLength() || pTagEntry->TagInfo.offset > pIO->GetLength() - pTagEntry->TagInfo.size)
     return false;
 
   icTagTypeSignature sigType;
diff --git a/IccProfLib/IccTagMPE.cpp b/IccProfLib/IccTagMPE.cpp
index 3ea01ec..4bf6b3f 100644
--- a/IccProfLib/IccTagMPE.cpp
+++ b/IccProfLib/IccTagMPE.cpp
@@ -1039,7 +1039,7 @@ bool CIccTagMultiProcessElement::Read(icUInt32Number size, CIccIO *pIO)
   icElemTypeSignature sigElem;
 
   for (i=0; i<m_nProcElements; i++) {
-    if (m_position[i].offset+m_position[i].size > size) {
+    if (m_position[i].size > size || m_position[i].offset > size - m_position[i].size) {
       return false;
     }
 
