--- a/IccProfLib/IccMpeBasic.cpp
+++ b/IccProfLib/IccMpeBasic.cpp
@@ -152,6 +152,9 @@
  ******************************************************************************/
 CIccFormulaCurveSegment &CIccFormulaCurveSegment::operator=(const CIccFormulaCurveSegment &seg)
 {
+  if (&seg == this)
+    return *this;
+
   if (m_params)
     free(m_params);
 
@@ -924,6 +927,9 @@
  ******************************************************************************/
 CIccSampledCurveSegment &CIccSampledCurveSegment::operator=(const CIccSampledCurveSegment &curve)
 {
+  if (&curve == this)
+    return *this;
+
   if (m_pSamples)
     free(m_pSamples);
 
@@ -1357,6 +1363,9 @@
 ******************************************************************************/
 CIccSingleSampledCurve &CIccSingleSampledCurve::operator=(const CIccSingleSampledCurve &curve)
 {
+  if (&curve == this)
+    return *this;
+
   if (m_pSamples)
     free(m_pSamples);
 
@@ -4953,6 +4962,9 @@
  ******************************************************************************/
 CIccMpeMatrix &CIccMpeMatrix::operator=(const CIccMpeMatrix &matrix)
 {
+  if (&matrix == this)
+    return *this;
+
   m_nReserved = matrix.m_nReserved;
 
   m_nInputChannels = matrix.m_nInputChannels;
@@ -6400,6 +6412,9 @@
 
 CIccMpeJabToXYZ &CIccMpeJabToXYZ::operator=(const CIccMpeJabToXYZ &cam)
 {
+  if (&cam == this)
+    return *this;
+
   if (cam.m_pCAM) {
     if (m_pCAM)
       delete m_pCAM;
@@ -6455,6 +6470,9 @@
 
 CIccMpeXYZToJab &CIccMpeXYZToJab::operator=(const CIccMpeXYZToJab &cam)
 {
+  if (&cam == this)
+    return *this;
+
   if (cam.m_pCAM) {
     if (m_pCAM)
       delete m_pCAM;
