diff --git a/IccProfLib/IccTagLut.cpp b/IccProfLib/IccTagLut.cpp
index 93739ad..58252e0 100644
--- a/IccProfLib/IccTagLut.cpp
+++ b/IccProfLib/IccTagLut.cpp
@@ -2494,7 +2510,16 @@ void CIccCLUT::Interp1d(icFloatNumber *destPixel, const icFloatNumber *srcPixel)
   icFloatNumber nu = (icFloatNumber)(1.0 - u);
 
   int i;
-  icFloatNumber *p = &m_pData[ix*n001];
+
+  int offset = ix*n001;
+
+  // clip offset to valid data address, in case of bad inputs
+  int maxDataSize = NumPoints()*m_nOutput;
+  int maxDataOffset = maxDataSize - (m_nOutput + n001);
+  if (offset > maxDataOffset)
+    offset = maxDataOffset;
+
+  icFloatNumber *p = &m_pData[offset];
 
   //Normalize grid units
   icFloatNumber dF0, dF1, pv;
@@ -2625,7 +2650,16 @@ void CIccCLUT::Interp3dTetra(icFloatNumber *destPixel, const icFloatNumber *srcP
   }
 
   int i;
-  icFloatNumber *p = &m_pData[ix*n001 + iy*n010 + iz*n100];
+
+  int offset = ix*n001 + iy*n010 + iz*n100;
+
+  // clip offset to valid data address, in case of bad inputs
+  int maxDataSize = NumPoints()*m_nOutput;
+  int maxDataOffset = maxDataSize - (m_nOutput + n111);
+  if (offset > maxDataOffset)
+    offset = maxDataOffset;
+
+  icFloatNumber *p = &m_pData[offset];
 
   //Normalize grid units
 
@@ -2723,7 +2757,16 @@ void CIccCLUT::Interp3d(icFloatNumber *destPixel, const icFloatNumber *srcPixel)
   icFloatNumber nu = 1.0f - u;
 
   int i;
-  icFloatNumber *p = &m_pData[ix*n001 + iy*n010 + iz*n100];
+
+  int offset = ix*n001 + iy*n010 + iz*n100;
+
+  // clip offset to valid data address, in case of bad inputs
+  int maxDataSize = NumPoints()*m_nOutput;
+  int maxDataOffset = maxDataSize - (m_nOutput + n111);
+  if (offset > maxDataOffset)
+    offset = maxDataOffset;
+
+  icFloatNumber *p = &m_pData[offset];
 
   //Normalize grid units
   icFloatNumber dF0, dF1, dF2, dF3, dF4, dF5, dF6, dF7, pv;
@@ -2812,7 +2855,16 @@ void CIccCLUT::Interp4d(icFloatNumber *destPixel, const icFloatNumber *srcPixel)
   icFloatNumber nv = 1.0f - v;
 
   int i, j;
-  icFloatNumber *p = &m_pData[iw*n001 + ix*n010 + iy*n100 + iz*n1000];
+
+  int offset = iw*n001 + ix*n010 + iy*n100 + iz*n1000;
+
+  // clip offset to valid data address, in case of bad inputs
+  int maxDataSize = NumPoints()*m_nOutput;
+  int maxDataOffset = maxDataSize - (m_nOutput + (int)m_nOffset[15]);
+  if (offset > maxDataOffset)
+    offset = maxDataOffset;
+
+  icFloatNumber *p = &m_pData[offset];
 
   //Normalize grid units
   icFloatNumber dF[16], pv;
@@ -2919,7 +2971,16 @@ void CIccCLUT::Interp5d(icFloatNumber *destPixel, const icFloatNumber *srcPixel)
   icFloatNumber ns4 = 1.0f - s4;
 
   int i, j;
-  icFloatNumber *p = &m_pData[ig0*n001 + ig1*n010 + ig2*n100 + ig3*n1000 + ig4*n10000];
+
+  int offset = ig0*n001 + ig1*n010 + ig2*n100 + ig3*n1000 + ig4*n10000;
+
+  // clip offset to valid data address, in case of bad inputs
+  int maxDataSize = NumPoints()*m_nOutput;
+  int maxDataOffset = maxDataSize - (m_nOutput + (int)m_nOffset[31]);
+  if (offset > maxDataOffset)
+    offset = maxDataOffset;
+
+  icFloatNumber *p = &m_pData[offset];
 
   //Normalize grid units
   icFloatNumber dF[32], pv;
@@ -3054,7 +3115,16 @@ void CIccCLUT::Interp6d(icFloatNumber *destPixel, const icFloatNumber *srcPixel)
   icFloatNumber ns5 = 1.0f - s5;
 
   int i, j;
-  icFloatNumber *p = &m_pData[ig0*n001 + ig1*n010 + ig2*n100 + ig3*n1000 + ig4*n10000 + ig5*n100000];
+
+  int offset = ig0*n001 + ig1*n010 + ig2*n100 + ig3*n1000 + ig4*n10000 + ig5*n100000;
+
+  // clip offset to valid data address, in case of bad inputs
+  int maxDataSize = NumPoints()*m_nOutput;
+  int maxDataOffset = maxDataSize - (m_nOutput + (int)m_nOffset[63]);
+  if (offset > maxDataOffset)
+    offset = maxDataOffset;
+
+  icFloatNumber *p = &m_pData[offset];
 
   //Normalize grid units
   icFloatNumber dF[64], pv;
@@ -3164,7 +3234,15 @@ void CIccCLUT::InterpND(icFloatNumber *destPixel, const icFloatNumber *srcPixel,
     index += ig[i]*m_DimSize[i];
   }
 
-  icFloatNumber *p = &m_pData[index];
+  int offset = (int)index;
+
+  // clip offset to valid data address, in case of bad inputs
+  int maxDataSize = NumPoints()*m_nOutput;
+  int maxDataOffset = maxDataSize - (m_nOutput + (int)m_nOffset[m_nNodes-1]);
+  if (offset > maxDataOffset)
+    offset = maxDataOffset;
+
+  icFloatNumber *p = &m_pData[offset];
   icFloatNumber temp[2];
   icFloatNumber pv;
   int nFlag = 0;
