diff --git a/IccProfLib/IccMpeBasic.cpp b/IccProfLib/IccMpeBasic.cpp
index a61a0d0..c4a702f 100644
--- a/IccProfLib/IccMpeBasic.cpp
+++ b/IccProfLib/IccMpeBasic.cpp
@@ -153,5 +153,8 @@ CIccFormulaCurveSegment::CIccFormulaCurveSegment(const CIccFormulaCurveSegment &
  ******************************************************************************/
 CIccFormulaCurveSegment &CIccFormulaCurveSegment::operator=(const CIccFormulaCurveSegment &seg)
 {
+  if (&seg == this)
+    return *this;
+
   free(m_params);
 
@@ -933,5 +936,8 @@ CIccSampledCurveSegment::CIccSampledCurveSegment(const CIccSampledCurveSegment &
  ******************************************************************************/
 CIccSampledCurveSegment &CIccSampledCurveSegment::operator=(const CIccSampledCurveSegment &curve)
 {
+  if (&curve == this)
+    return *this;
+
   free(m_pSamples);
 
@@ -1367,5 +1373,8 @@ CIccSingleSampledCurve::CIccSingleSampledCurve(const CIccSingleSampledCurve &cur
 ******************************************************************************/
 CIccSingleSampledCurve &CIccSingleSampledCurve::operator=(const CIccSingleSampledCurve &curve)
 {
+  if (&curve == this)
+    return *this;
+
   free(m_pSamples);
 
