diff --git a/IccProfLib/IccMpeBasic.cpp b/IccProfLib/IccMpeBasic.cpp
index 7d2a7a1..e98b972 100644
--- a/IccProfLib/IccMpeBasic.cpp
+++ b/IccProfLib/IccMpeBasic.cpp
@@ -2199,7 +2199,7 @@ void CIccSampledCalculatorCurve::Describe(std::string &sDescription, int nVerbos
     snprintf(buf, bufSize, "%.8f,", m_lastEntry);
     sDescription += buf;
 
-    snprintf(buf, bufSize, "%d", m_nDesiredSize);
+    snprintf(buf, bufSize, "%u", m_nDesiredSize);
     sDescription += buf;
 
     snprintf(buf, bufSize, "]\r\n");
@@ -2214,7 +2214,7 @@ void CIccSampledCalculatorCurve::Describe(std::string &sDescription, int nVerbos
     snprintf(buf, bufSize, "%.8f,", m_lastEntry);
     sDescription += buf;
 
-    snprintf(buf, bufSize, "%d", m_nDesiredSize);
+    snprintf(buf, bufSize, "%u", m_nDesiredSize);
     sDescription += buf;
 
     snprintf(buf, bufSize, "]\r\n");
@@ -5388,7 +5388,7 @@ static icFloatNumber NoClip(icFloatNumber v)
 }
 
 // this isn't used yet, but want to have it ready
-static icFloatNumber RealUnitClip(icFloatNumber v)
+[[maybe_unused]] static icFloatNumber RealUnitClip(icFloatNumber v)
 {
   if (std::isnan(v))
     return icFloatNumber(0);
