diff --git a/IccProfLib/IccCmm.cpp b/IccProfLib/IccCmm.cpp
index 82d5561..3ef8cdd 100644
--- a/IccProfLib/IccCmm.cpp
+++ b/IccProfLib/IccCmm.cpp
@@ -1458,11 +1458,15 @@ icStatusCMM CIccXform::Begin()
       // scale factors depend upon media white point
       // set up for input transform
       if (!m_bInput) {
+        if (mediaXYZ[0] == 0.0f || mediaXYZ[1] == 0.0f || mediaXYZ[2] == 0.0f)
+          return icCmmStatInvalidProfile;
         m_PCSScale[0] = illumXYZ[0] / mediaXYZ[0];
         m_PCSScale[1] = illumXYZ[1] / mediaXYZ[1];
         m_PCSScale[2] = illumXYZ[2] / mediaXYZ[2];
       }
       else {
+        if (illumXYZ[0] == 0.0f || illumXYZ[1] == 0.0f || illumXYZ[2] == 0.0f)
+          return icCmmStatInvalidProfile;
         m_PCSScale[0] = mediaXYZ[0] / illumXYZ[0];
         m_PCSScale[1] = mediaXYZ[1] / illumXYZ[1];
         m_PCSScale[2] = mediaXYZ[2] / illumXYZ[2];
