--- a/IccProfLib/IccTagLut.h	2026-03-26 22:13:40.592048435 -0400
+++ IccProfLib/IccTagLut.h	2026-03-26 22:13:40.600070241 -0400
@@ -640,12 +640,12 @@
     return m_NumberOfTriangles;
   }	
 
-  icInt16Number getNumPCSChannels() 
+  icUInt16Number getNumPCSChannels() 
   {
     return m_nPCSChannels;
   }
 
-  icInt16Number getNumDeviceChannels() 
+  icUInt16Number getNumDeviceChannels() 
   {
     return m_nDeviceChannels;
   }		
@@ -653,8 +653,8 @@
 protected:
   icInt32Number m_NumberOfVertices;
   icInt32Number m_NumberOfTriangles;
-  icInt16Number m_nPCSChannels;
-  icInt16Number m_nDeviceChannels;
+  icUInt16Number m_nPCSChannels;
+  icUInt16Number m_nDeviceChannels;
   icFloatNumber* m_PCSValues;
   icFloatNumber* m_DeviceValues;
   icGamutBoundaryTriangle* m_Triangles;
