diff --git a/IccProfLib/IccTagLut.cpp b/IccProfLib/IccTagLut.cpp
index 444f654..3389c41 100644
--- a/IccProfLib/IccTagLut.cpp
+++ b/IccProfLib/IccTagLut.cpp
@@ -906,3 +906,3 @@ void CIccTagParametricCurve::Describe(std::string &sDescription, int /* nVerbose
-  snprintf(buf, bufSize, "Y = %lf * X   when (X < %.4lf)\n",
+  snprintf(buf, bufSize, "Y = %.4lf * X   when (X < %.4lf)\n",
     m_dParam[3], m_dParam[4]);
   sDescription += buf;
@@ -917,3 +917,3 @@ void CIccTagParametricCurve::Describe(std::string &sDescription, int /* nVerbose
-  snprintf(buf, bufSize, "Y = %lf * X + %.4lf  when (X < %.4lf)\n",
+  snprintf(buf, bufSize, "Y = %.4lf * X + %.4lf  when (X < %.4lf)\n",
     m_dParam[3], m_dParam[6], m_dParam[4]);
   sDescription += buf;
