diff --git a/Tools/CmdLine/IccApplyProfiles/TiffImg.cpp b/Tools/CmdLine/IccApplyProfiles/TiffImg.cpp
--- a/Tools/CmdLine/IccApplyProfiles/TiffImg.cpp
+++ b/Tools/CmdLine/IccApplyProfiles/TiffImg.cpp
@@ -105 +105 @@ static bool IsSpaceChar(icUInt32Number sig)
-const icUInt16Number kMaxTiffSamples = std::numeric_limits<icUInt16Number>::max();
+const icUInt16Number kMaxTiffSamples = 32;
@@ -445 +445 @@ bool CTiffImg::Open(const char *szFname)
-      m_nSamples == 0 ||
+      m_nSamples == 0 || m_nSamples > kMaxTiffSamples ||
@@ -544 +544 @@ bool CTiffImg::ReadLine(unsigned char *pBuf)
-      m_nSamples == 0 ||
+      m_nSamples == 0 || m_nSamples > kMaxTiffSamples ||
