diff --git a/IccProfLib/IccApplyBPC.cpp b/IccProfLib/IccApplyBPC.cpp
--- a/IccProfLib/IccApplyBPC.cpp
+++ b/IccProfLib/IccApplyBPC.cpp
@@ -427,6 +427,11 @@ bool CIccApplyBPC::calcDstBlackPoint(const CIccProfile* pProfile, const CIccXfor
 		}
 		pcs2lab(Pixel, pProfile);
 		icFloatNumber MaxL = Pixel[0];
+		icFloatNumber LRange = MaxL - MinL;
+		if (!std::isfinite(LRange) || fabs(LRange) < 1.0e-6f) {
+			delete pCmm;
+			return false;
+		}
 
 		// check if quadratic estimation needs to be done
 		bool bStraightMidRange = false;
@@ -459,7 +464,7 @@ bool CIccApplyBPC::calcDstBlackPoint(const CIccProfile* pProfile, const CIccXfor
 				pcs2lab(Pixel, pProfile);
 				roundtripL = Pixel[0];
 
-				if (roundtripL>(MinL + 0.2 * (MaxL - MinL))) {
+				if (roundtripL>(MinL + 0.2 * LRange)) {
 					if (fabs(roundtripL - lcnt)>4.0) {
 						bStraightMidRange = false;
 						break;
@@ -502,7 +507,7 @@ bool CIccApplyBPC::calcDstBlackPoint(const CIccProfile* pProfile, const CIccXfor
 				return false;
 			}
 			pcs2lab(Pixel, pProfile);
-			y[i] = (Pixel[0] - MinL)/(MaxL - MinL);
+			y[i] = (Pixel[0] - MinL)/LRange;
 		}
 
 		// check for y values in the range and rearrange
